### PR TITLE
openldap: added the package

### PR DESCRIPTION
**Pull request by sophieeihpos**
Specify library name and version:  **openldap/2.6.0**

Our internal libraries depends on openldap.

© 2022 Morgan Stanley
THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE APACHE LICENSE V.2.0. YOU MAY OBTAIN A COPY OF THE LICENSE AT https://www.apache.org/licenses/LICENSE-2.0.
THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/openldap/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        autotools.libs.remove("pthread") 
+        for lib in autotools.libs:
+            libs += "-l{} ".format(lib)
+        libs += "-pthread "
+        with tools.environment_append({"systemdsystemunitdir": self.package_folder, "LD_LIBRARY_PATH": libpath, "LIBS": libs}):
+            autotools.configure(
+                configure_dir=os.path.join(
+                    self._source_subfolder),
+                args=conf_args)
+        self.run("make depend -j8")
```
Comment (SSE4) [inline recipes/openldap/all/conanfile.py:77]: better use `tools.cpu_count()`

Comment (sophieeihpos) [reply recipes/openldap/all/conanfile.py]: Updated. Thanks.

**Comment by uilianries** [comment]
@sophieeihpos By checksum reported, you are trying to build the version 2.6.0, but latest version is 2.6.1, which corresponds to the checksum 9d576ea6962d7db8a2e2808574e8c257c15aef55f403a1fb5a0faf35de70e6f3

**Review by jgsogo — DISMISSED** [review]
LGTM

File: recipes/openldap/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+add_executable(${PROJECT_NAME} test_package.cpp)
+# For gcc command, the -lA must come before -lB for static libs, if A depends on B.
+list(REMOVE_ITEM CONAN_LIBS "ldap")
+list(INSERT CONAN_LIBS 0 "ldap")
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/openldap/all/test_package/CMakeLists.txt:9]:
```suggestion
conan_basic_setup(TARGETS)

find_package(openldap CONFIG REQUIRED)

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} openldap::openldap)
```

As I suggested on https://github.com/sophieeihpos/conan-center-index/pull/1, this linker order must be solved by the cmake file generated by Conan

File: recipes/openldap/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            configure_dir=self._source_subfolder,
+            vars=self._configure_vars)
+        return self._autotools
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        autotools = self._configure_autotools()
+
+        autotools.make(args=["depend"], vars=self._configure_vars)
```
Comment (uilianries) [inline recipes/openldap/all/conanfile.py:93]:
```suggestion
```

Not needed, only more CI time wasted. All dependencies are already solved, don't need to scan to check it

File: recipes/openldap/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+    def requirements(self):
+        self.requires("openssl/1.1.1m")
```
Comment (uilianries) [inline recipes/openldap/all/conanfile.py:43]:
```suggestion
        self.requires("openssl/1.1.1n")
```
Between the interval of your first commit and the last, a new security patch of OpenSSL was released

File: recipes/openldap/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            "--with-cyrus_sasl={}".format(yes_no(self.options.with_cyrus_sasl)),
+            "--with-pic={}".format(yes_no(self.options.get_safe("fPIC", True))),
+            "--without-fetch",
+            "--with-tls=openssl",
+            "--enable-auditlog"]
+        self._configure_vars = self._autotools.vars
+        self._configure_vars["systemdsystemunitdir"] = os.path.join(
+            self.package_folder, "res")
+
+        # Need LD_LIBRARY_PATH for configure. Otherwise, configure fails.
```
Comment (uilianries) [inline recipes/openldap/all/conanfile.py:70]: For what configuration do you need it? I still don't see error without LD_LIBRARY_PATH

File: recipes/openldap/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        libpath = ""
+        for path in self._autotools.library_paths:
+            libpath += "{}:".format(path)
+        self._configure_vars["LD_LIBRARY_PATH"] = libpath
+
```
Comment (uilianries) [inline recipes/openldap/all/conanfile.py:75]:
```suggestion
```
For what configuration do you need it? I still don't see error without LD_LIBRARY_PATH

**Review by jgsogo — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**